src/RestApiClientSharp/RestApiClient.Events.cs

```diff
@@ -8,39 +8,83 @@ public partial class RestApiClient : ObservableObject, IRestApiClient
 
         #region EventHandlers
         public event EventHandler? Error;
+        public bool ReThrowOnError;
+
         protected virtual void OnError()
         {
             Error?.Invoke(this, EventArgs.Empty);
         }
         protected virtual void OnError(UnhandledExceptionEventArgs e)
         {
-            Error?.Invoke(this, e);
+            if (ReThrowOnError)
+            {
+                throw (Exception)e.ExceptionObject;
+            }
+            else
```

Comment:
I would prefer this way. In my opinion, the `Error` event should be triggered every time, even if the exception is re-thrown. What do you think?

```cs
Error?.Invoke(this, e);
if (ReThrowOnError)
{
    throw e.Exception ?? throw new Exception(e.Message);
}
```